File: C_Programming/Competitive/program28.c
//////  Write a program to find factorial of a given number.

//////  Input : 5
//////  Output : 120

//////  Input : -5
//////  Output : 120

//////  Input : 4
//////  Output : 24

#include<stdio.h>

int Factorial(int iNo)
{
    int iCnt = 0;
    int iFact = 1;

    if (iNo < 0){
        iNo = -iNo;
    }

    for (iCnt = 1; iCnt <= iNo; iCnt++)
    {
        iFact = iFact * iCnt;
    }

    return iFact;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter a number : \n");
    scanf("%d", &iValue);

    iRet = Factorial(iValue);

    printf("%d\n", iRet);
    
    return 0;
}

File: Java_Programming/Competitive/program28.java
////    Write a program which accepts a number from user and toggle 7th bit of that number.
///     Return modified number.
///     Input : 137
///     Output : 201

import java.util.*;

class program28
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        int iValue = 0;
        int iRet = 0;

        System.out.println("Enter a number : ");
        iValue = sobj.nextInt();

        Bitwise bobj = new Bitwise();

        iRet = bobj.ToggleBit(iValue);

        System.out.println("Modified number is : " + iRet);

        sobj.close();
    }
}

class Bitwise
{
    int ToggleBit(int iNo)
    {
        int iMask = 0x40;

        return (iNo ^ iMask);
    }
}